pkg/console/status/auth_status.go

```diff
@@ -94,69 +94,133 @@ func (c *AuthStatusHandler) Apply(ctx context.Context, authnConfig *configv1.Aut
 	if err != nil {
 		return err
 	}
+	existingClientStatusIndex, existingClientStatus := c.getExistingClientStatus(applyConfig)
```

Comment:
This looks incorrect, you shouldn't need to worry about any indices at all.

Use server-side apply so that you don't have to care about any indices. Pass a controller name to `AuthStatusHandler`. Each controller is taking care of their own client.

You can see that the slice is ready to be handled that way if you look at the API: https://github.com/openshift/api/blob/master/config/v1/types_authentication.go#L115-L117

Read more about SSA at https://kubernetes.io/docs/reference/using-api/server-side-apply/ and/or ask me.

The diff looks much bigger than I expected given we only want to parametrize componentName/componentNamespace. Is it because the above principle was not well understood prior to writing the code?

Reply:
Yes, I had a limited understanding of SSA. My first pass at this was a much smaller change, but caused an infinite loop. I didn't realize the cause, which you pointed out, is the duplicate field manager.